tracer/src/Datadog.Trace/TracerManager.cs

```diff
@@ -682,13 +682,29 @@ private static TracerManager CreateInitializedTracer(TracerSettings settings, Tr
             return newManager;
         }
 
-        private static void OneTimeSetup()
+        private static void OneTimeSetup(TracerSettings tracerSettings)
         {
             // Register callbacks to make sure we flush the traces before exiting
             LifetimeManager.Instance.AddAsyncShutdownTask(RunShutdownTasksAsync);
 
             // start the heartbeat loop
             _heartbeatTimer = new Timer(HeartbeatCallback, state: null, dueTime: TimeSpan.Zero, period: TimeSpan.FromMinutes(1));
+
+            if (FrameworkDescription.Instance.OSPlatform == OSPlatformName.Linux && Environment.Is64BitProcess && !Util.EnvironmentHelpers.IsServerlessEnvironment())
```

Comment:
We should potentially think about how this is rolling out 🤔 For data pipeline, we very explicitly _don't_ want to start using libdatadog everywhere at once, because we don't know for sure what the impact will be. Maybe it's not too bad as it's linux only,  specifically not serverless, and only a single call. It might act as a decent test essentially 🤔

Reply:
I mean, we explicitly _didn't_ want to roll this out to linux first, given that it will impact everyone using SSI, but I guess that's what we're doing now 😬